itest.py

```diff
@@ -0,0 +1,832 @@
+
+    os_of_choice = IRODS_TEST_PROJECTS_DIR / os_choice
+    mid = map(lambda d: d.name, filter(lambda d: d.is_dir(), os_of_choice.iterdir()))
+    db_choices = list(map(lambda d: d[len(os_choice)+1:], mid))
+
+    if db_choice is None:
+        for db in db_choices:
+            print(db)
+
+        db_choice = input("What db ya want? ")
+
+    validate_choice(db_choice, db_choices)
+
+    return ((os_of_choice / f'{os_choice}-{db_choice}'), os_choice, db_choice)
+
+def query_desired_test(test_choice):
+    test_files = list(map(lambda f: f.stem, IRODS_TEST_DIR.glob('run_*_tests.py')))
+
+    if test_choice is None:
+        for f in test_files:
+            print(f)
+
+        test_choice = input("What test ya want? ")
+
+    validate_choice(test_choice, test_files)
+    return IRODS_TEST_DIR / f'{test_choice}.py'
+
+# -------
+# Runners
+# -------
+def ask_run_test(os=None, db=None, test_type=None, test_args=None):
+    desired_test = query_desired_test(test_type)
+    project_directory, os_choice, _ = query_desired_test_project(os, db)
+
+    # Build up named args
+    named_args = [('--project-directory', project_directory),
+                  ('--irods-package-directory', IRODS_PACKAGES_DIR),
+                  ('--output-directory', IRODS_LOGS_DIR),]
+    bonus_args = []
+
+    # python run_plugin_tests.py --project-directory projects/ubuntu-22.04/ubuntu-22.04-postgres-14.8/ --irods-package-directory ../build-artifacts/ubuntu-22/packages/ --plugin-package-directory ../build-artifacts/ubuntu-22/plugins --discard-logs irods_microservice_plugins_curl
+    # match/case candidate?
+    if test_args is None:
+        if desired_test.stem == 'run_plugin_tests':
+            # Gather valid plugins
+            yes_plugins = tuple(get_supported_plugins())
+            assert(len(yes_plugins) > 0)
+
+            for plugin in yes_plugins:
+                print(plugin)
+    
+            # Prompt user for desired plugin
+            plugin_choice = input("What plugin ya wanna build? ")
+            validate_choice(plugin_choice, yes_plugins)
+
+            # Add plugin git name
+            bonus_args.append(plugin_choice)
+
+            # Dumb trim of plugin
+            for prefix in PLUGIN_PREFIXES:
+                if plugin_choice.startswith(prefix):
+                    plugin_name = plugin_choice[len(prefix)+1:]
+
+            # Append plugin dir
+            named_args.append(('--plugin-package-directory', str(IRODS_PLUGIN_PACKAGE_DIR).format(os_name=os_choice, short_plugin_name=plugin_name)))
+        elif desired_test.stem == 'run_core_tests':
+            num_executors = int(input('How many test executors do you want? '))
+            named_args.append(('--concurrent-test-executor-count', str(num_executors)))
+        elif desired_test.stem == 'run_topology_tests':
+            num_executors = int(input('How many test executors do you want? '))
+            named_args.append(('--concurrent-test-executor-count', str(num_executors)))
```

Comment:
Noting here, but it applies elsewhere, we should separate the interactive part so the validation/work can be entirely separate from the prompts.

Reply:
This will be made into a separate issue.